docs/content/examples/data_engineers.md

```diff
@@ -6,596 +6,309 @@ date: 2023-05-17T17:03:41-04:00
+
+```python
+import os
+import json
+import asyncio
+from datetime import datetime
+
+import requests
+from pyensign.events import Event
+from pyensign.ensign import Ensign
+
+class WeatherPublisher:
+    def __init__(self, topic="current-weather", location="Washington, DC"):
+        """
+        Create a publisher that publishes weather events for a location to a topic.
+        """
+        self.topic = topic
+        self.location = location
+        self.weather_api_key = os.environ.get("WEATHER_API_KEY")
+        self.ensign = Ensign()
 ```
-### Start the First Stream
-
-Next we want to create a long-running process that will continue pinging the Weather API, parsing the response, and publishing it for downstream consumption.
-
-In our `main()` function, we will first create a logger using `watermill.NewStdLogger`. Then we create the publisher and create the `closeCh` channel that will be used to send a signal to the publisher to stop publishing. We then pass the `publisher` and the `closeCh` to the `publishWeatherData` function and run it in a goroutine.
 
-We then create another channel that listens for a `os.Interrupt` signal and will close when it receives the signal. If it receives the signal (e.g. because we want to stop the process, or if something goes wrong), it closes and the code moves on to close the `closeCh` channel and that notifies the publisher to stop publishing.
+_Pro Tip: Calling Ensign() will automatically load your client ID and client secret from the environment_
 
-```golang
-func main() {
-	// add a logger
-	logger := watermill.NewStdLogger(false, false)
-	logger.Info("Starting the producer", watermill.LogFields{})
+### Receive and Publish
 
-	//create the publisher
-	publisher, err := ensign.NewPublisher(
-		ensign.PublisherConfig{
-			Marshaler: ensign.EventMarshaler{},
-		},
-		logger,
-	)
-	if err != nil {
-		panic(err)
-	}
-	defer publisher.Close()
+Most publishers follow the wait-and-publish pattern. They do a lot of waiting, and then occasionally publish one or more events when something happens (e.g. a timer expires or an asynchronous signal is received). In Python, this usually looks like a coroutine with a loop.
 
-	//used to signal the publisher to stop publishing
-	closeCh := make(chan struct{})
+```python
+    async def recv_and_publish(self):
+		"""
+        Receive weather events and publish them to the topic.
+        """
 
-	go publishWeatherData(publisher, closeCh)
+        # Ensure the topic exists
+        await self.ensign.ensure_topic_exists(self.topic)
 
-	// wait for SIGINT - this will end processing
-	c := make(chan os.Signal, 1)
-	signal.Notify(c, os.Interrupt)
-	<-c
+        while True:
+            # Make a request to the weather API
+            response = requests.get("http://api.weatherapi.com/v1/current.json", params={
+                "key": self.weather_api_key,
+                "q": self.location,
+            })
+            response.raise_for_status()
```

Comment:
I noticed in the next code snippet, there is try, except logic around this statement.  Add that here as well for consistency.